tx/src/entities.rs

```diff
@@ -23,20 +23,21 @@ use self::edn::symbols::NamespacedKeyword;
 pub enum TempId {
     External(String),
     Internal(i64),
+    Tx, // Special identifier used to refer to the current transaction.
 }
 
 impl TempId {
     pub fn into_external(self) -> Option<String> {
         match self {
             TempId::External(s) => Some(s),
-            TempId::Internal(_) => None,
+            _ => None,
         }
     }
 
     pub fn into_internal(self) -> Option<i64> {
         match self {
             TempId::Internal(x) => Some(x),
-            TempId::External(_) => None,
+            _ => None,
```

Comment:
The trick you used to find these locations — add the `enum` case, look for the compile errors — stops working if you use `_`. Add the explicit `TempId::Tx` case here instead:

```
TempId::Internal(x) => Some(x),
TempId::Tx |
TempId::External(_) => None,
```